truss/templates/control/control/helpers/inference_server_controller.py

```diff
@@ -16,11 +17,16 @@ class InferenceServerController:
     delegates to InferenceServerProcessController.
     """
 
+    _process_controller: InferenceServerProcessController
+    _patch_applier: PatchApplier
+    _app_logger: logging.Logger
+    _oversee_inference_server: bool
```

Comment:
Curios, why define these class variables, here and in other classes in this PR.

Reply:
I think having unused class variables makes the code confusing,  and does not seem ideal, I don't feel the tradeoff is worth it. Can you try typing these in `__init__` and see if IDE recognizes them.

Reply:
So, these aren't unused class variables, they are instance variable type hints. The two options here are typing out the `__init__` method parameters or adding them in the class body.   I don't mind hugely them being on in `__init__`, it just makes readability much more difficult when I'm trying to ascertain instance variables, especially when the `__init__` methods are complex and aren't simple instance variable setting.  IDE detection is more a bonus. 

FWIW, Guido seems to prefer this method as it might be "the future" of the language: https://gitter.im/python/typing?at=595f9f5b2723db8d5ee9c892

This is more a philosophical cleanliness detail, but not a hill I care to die on if you feel strongly.

Reply:
If these indeed aren't class variables, then this seems fine. I saw somewhere this being alluded to but could not find clear documentation on it.